keps/sig-node/24-apparmor-ga/README.md

```diff
@@ -0,0 +1,682 @@
+node upgrade and downgrade. No special handling or consideration is needed to
+support this.
+
+On the API server side, we've already taken version skew in HA clusters into
+account. The same precautions make upgrade & downgrade handling a non-issue.
+
+Since
+[we support](https://kubernetes.io/docs/setup/release/version-skew-policy/) up
+to 2 minor releases of version skew between the master and node, annotations
+must continue to be supported and backfilled for at least 2 versions passed the
+initial implementation. However, we can decide to extend support farther to
+reduce breakage. If this feature is implemented in v1.20, I propose v1.24 as a
+target for removal of the old behavior.
+
+### Test Plan
+
+AppArmor already has [e2e tests][https://github.com/kubernetes/kubernetes/blob/6596a14/test/e2e_node/apparmor_test.go],
+but the tests are guarded by the `[Feature:AppArmor]` tag and not run in the
+standard test suites.
+
+Prior to being marked GA, the feature tag will be removed from the AppArmor
+tests, and the tests will be migrated to the new fields API. Tests will be
+tagged as `[LinuxOnly]`.
+
+New tests will be added covering the annotation/field conflict cases described
+under [Version Skew Strategy](#version-skew-strategy).
+
+Test coverage for localhost profiles will be added, unless we decide to [keep
+localhost support in alpha](#alternatives).
+
+### Graduation Criteria
+
+_This section is excluded, as it is the subject of the entire proposal._
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable, can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/20190731-production-readiness-review-process.md
+
+Production readiness review questionnaire must be completed for features in
+v1.19 or later, but is non-blocking at this time. That is, approval is not
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature enablement and rollback
+
+_This section must be completed when targeting alpha to a release._
+
+- **How can this feature be enabled / disabled in a live cluster?**
+
+  This feature was implemented prior to the concept of feature gates. Meaning,
+  it is already enabled in live clusters and cannot be disabled. This KEP is
+  simply "cleaning up" the API to make it GA.
```

Comment:
is this accurate? disabling the AppArmor feature drops the annotations when creating a pod today - see https://github.com/kubernetes/kubernetes/blob/7705fa295c6470a3e9b34ddad4f5304ed8058d88/pkg/api/pod/util.go#L364-L370

Reply:
Ah yeah, looks like it was one of the first features…changed the paragraph to:
```
  The feature can be enabled/disabled by the `AppArmor` feature gate. This
  behavior will not change when moving to GA.
```